.github/workflows/release-build.yml

```diff
@@ -28,6 +28,24 @@ jobs:
       - name: Checkout Contrib Repo @ SHA - ${{ github.sha }}
         uses: actions/checkout@v4
 
+      - name: Check main build status
+        run: |
+          WORKFLOW_ID=$(gh api repos/${{ github.repository }}/actions/workflows --jq '.workflows[] | select(.name=="NodeJS Instrumentation Main Build") | .id')
+          LATEST_RUN=$(gh api repos/${{ github.repository }}/actions/workflows/$WORKFLOW_ID/runs --jq '.workflow_runs[] | select(.head_branch=="${{ github.ref_name }}") | {conclusion, status}' | head -1)
```

Comment:
One assumption here is that this is ordered by date, which is likely true, but could(?) change in the future. Can you also sort the array from `workflow_runs[]` by the workflow date so guarantee the ordering so we can query the latest?

Reply:
good catch, updated in all 3 PRs. see latest commit for diff in jq command.